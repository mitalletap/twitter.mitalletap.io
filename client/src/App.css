/* Root */
:root {
  --primary-color: #163460;
  --secondary-color: #63cdda ;
  --background-default: #021C33;
}

h1 {
  color: #a1afb2 !important;
  font-weight: bold;
}

/* App */

.app-container, .app-content, .app-center-content {
  height: 100%;
  background-color:var(--primary-color);
} 



/* NavBar */

.sider-navbar {
  background-color:var(--primary-color) !important;
}

.navbar-container {
  height: 100%;
}

.navbar-menu {
  height: 100% !important;
}

.navbar-menu-item-avatar {
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.navbar-menu-item-profile {
  height: 100px !important;
}


.home-content {
  background-color: var(--primary-color);
  padding-top: 100px;
  padding-bottom: 20px;
}

.home-grid-row {
  margin: 0px !important;
}

.home-grid-col {
  /* background-color: dodgerblue; */
  margin: 10px;
}

.post-item-container {
  border-radius: 25px;
  padding: 20px;
  display: block;
  height: 100%;
  background-color: var(--background-default);
}

.post-item-time {
  display: block;
  margin-top: 20px;
}

.post-item-message {
  display: inline-block;
  width: 90%;
}

.post-item-username {
  display: block;
  float: right;
  color: #a1afb2 !important;
} 