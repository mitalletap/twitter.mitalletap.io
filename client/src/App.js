import React, { Component } from 'react';
import Amplify from 'aws-amplify';
import { withAuthenticator } from 'aws-amplify-react';
import { Layout, Menu, Input, Button } from 'antd';
import aws_exports from './aws-exports';
import Home from './components/Home';
import Post from './components/Post';
import NavBar from './components/NavBar';
import Friends from './components/Friends';
import Notifications from './components/Notifications';
import Profile from './components/Profile';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actionCreators from './actions/rootAction';

import './App.css';
import '@aws-amplify/ui/dist/style.css';
import 'antd/dist/antd.css';

Amplify.configure(aws_exports);
const { Header, Content, Footer, Sider } = Layout;

const signUpConfig = {
  hiddenDefaults: ["username"],
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: "Name",
      key: 'name',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: "Username",
      key: 'preferred_username',
      required: true,
      displayOrder: 2,
      type: 'string'
    },
    {
      label: "Data of Birth",
      key: 'birthdate',
      required: true,
      displayOrder: 3,
      type: 'date'
    }
  ]
}


export class App extends Component {
  constructor(props){
    super(props);
  }
  
  componentDidMount() {
    this.props.loadUserData();
  }

  render() { 
    return (  
      <div className="app-container">
        <Router>
            <Layout className="app-content">
              <Sider breakpoint="lg" collapsedWidth="auto" className="sider-navbar" theme="dark" width={200}>
                  <NavBar {...this.props} />
              </Sider>
              <Layout className="app-center-content">
                <Layout>
                  <Content>
                    <Switch>
                      <Route exact path="/"><Home {...this.props} /></Route>
                      <Route exact path="/friends"><Friends /></Route>
                      <Route exact path="/profile"><Profile /></Route>
                      <Route exact path="/notifications"><Notifications /></Route>
                      <Route exact path="/create-post"><Post {...this.props} /></Route>
                      <Route exact path="/u/:username"><Home /></Route>
                    </Switch>
                  </Content>
                </Layout>
              </Layout>
            </Layout>
          </Router>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return state;
}


// export default withAuthenticator(App, { signUpConfig });
export default withAuthenticator(connect(mapStateToProps, actionCreators)(App), {signUpConfig});