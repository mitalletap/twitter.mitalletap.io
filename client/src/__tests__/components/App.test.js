import React from 'react';
import { shallow, mount } from 'enzyme';
import { App } from '../../App';

describe('should test app page', () => {

  it('should match snapshot', () => {
    let app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

});