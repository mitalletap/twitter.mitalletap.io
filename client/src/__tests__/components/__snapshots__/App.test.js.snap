// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test app page should match snapshot 1`] = `ShallowWrapper {}`;
